tensorflow-core/tensorflow-core-api/src/main/java/org/tensorflow/Session.java

```diff
@@ -135,6 +143,111 @@ public void close() {
     }
   }
 
+  /**
+   * The result of a run in a session.  Contains the fetched tensors and the outputs that were fetched.
+   * <p>
+   * Closing a {@code Result} object will close all of the tensors contained by it.
+   */
+  public final class Result implements AutoCloseable, Iterable<Tensor<?>>{
```

Comment:
Should we fold in the `Run` class into `Result`, and make the metadata an optional field? Seems a little odd to have two different ways of getting it, one of which isn't `AutoCloseable`.

Reply:
Yeah, makes sense.